### PR TITLE
Migrate oidcauthextension from Collector core to Collector contrib

### PR DESCRIPTION
**Pull request by Saber-W**
**Description:**
This PR migrates the oidcauthextension component from the Collector core repository to the Collector contrib repository.

This PR retains the commit history from core and includes plumbing fixes for go.mod, makefile, components.go.

**Link to tracking Issue:**
[opentelemetry-collector/#3474](https://github.com/open-telemetry/opentelemetry-collector/issues/3474)
[opentelemetry-collector/#3652](https://github.com/open-telemetry/opentelemetry-collector/issues/3652)

### CODE REVIEW

**Comment by alolita** [comment]
Hi @bogdandrutu @tigrannajaryan based on our discussion last week, we're starting to file PRs to move each core component to contrib. Please review and merge. Thanks!

**Review by anuraaga — APPROVED** [review]

**Review by punya — APPROVED** [review]

**Review by bogdandrutu — CHANGES_REQUESTED** [review]
PR should show the files moved not new files

File: extension/oidcauthextension/go.mod [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module github.com/open-telemetry/opentelemetry-collector-contrib/extension/oidcauthextension
+
+go 1.16
+
+require (
+	github.com/coreos/go-oidc v2.2.1+incompatible
+	github.com/stretchr/testify v1.7.0
+	go.opentelemetry.io/collector v0.33.0
```
Comment (Aneurysm9) [inline extension/oidcauthextension/go.mod:8]: Should this depend on the same collector version the rest of the repo is pinned to?

Comment (Saber-W) [reply extension/oidcauthextension/go.mod:8]: yes, I will change this

**Review by bogdandrutu — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

File: extension/oidcauthextension/go.mod [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module github.com/open-telemetry/opentelemetry-collector-contrib/extension/oidcauthextension
+
+go 1.16
+
+require (
+	github.com/coreos/go-oidc v2.2.1+incompatible
+	github.com/stretchr/testify v1.7.0
+	go.opentelemetry.io/collector v0.33.1-0.20210820002854-d3000232f8f6
```
Comment (bogdandrutu) [inline extension/oidcauthextension/go.mod:8]: Add a `go.opentelemetry.io/collector/model v0.33.1-0.20210820002854-d3000232f8f6  // indirect`

Comment (bogdandrutu) [reply extension/oidcauthextension/go.mod:8]:
```
	go.opentelemetry.io/collector/model v0.33.1-0.20210820002854-d3000232f8f6 // indirect
```

Comment (bogdandrutu) [reply extension/oidcauthextension/go.mod:8]: Then run `go mod tidy` in this dir

Comment (Saber-W) [reply extension/oidcauthextension/go.mod:8]: okay - will do

**Pull request merged**